### PR TITLE
feature: init load more best selling food and drink

### PR DESCRIPTION
**Pull request by at-sonvu**

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 1]
```diff
@@ -91,7 +91,8 @@ final class HomeViewController: ViewController {
     }
 
     @IBAction func loadMoreFoodButtonTouchUpInside(_ sender: UIButton) {
-        let vc = SuggestionViewController()
+        let vm = SuggestionViewModel(type: .Food)
+        let vc = SuggestionViewController(viewModel: vm, type: .Suggestion, nibName: "SuggestionViewController", bundle: nil)
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: change, 
let vc = SuggestionViewController.vc()
vc.viewModel = viewmodel.suggesttionViewModelWith(type: .Food)
cái type: .Suggestion cho vào viewmodel luôn!

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 2]
```diff
@@ -163,3 +165,21 @@ extension HomeViewController: UIPageViewControllerDelegate {
         viewModel.pageViewController(pageViewController, currentPage: &currentPage)
     }
 }
+
+extension HomeViewController: HomeFooterSectionViewDelegate {
+    func footerSectionView(_ view: HomeFooterSectionView, _ selectBuyedType: ListMenuViewModel.MenuType) {
+        switch selectBuyedType {
+        case .MostBuyedDrink:
+            let vm = SuggestionViewModel(type: .MostBuyedDrink)
+            let vc = SuggestionViewController(viewModel: vm, type: .MostBuyed, nibName: "SuggestionViewController", bundle: nil)
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: như trên

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 3]
```diff
@@ -163,3 +165,21 @@ extension HomeViewController: UIPageViewControllerDelegate {
...
+            vc.drinkButtonleadingConstraint = UIScreen.main.bounds.width / 2
+            navigationController?.pushViewController(vc, animated: true)
+        case .MostBuyedFood:
+            let vm = SuggestionViewModel(type: .MostBuyedFood)
+            let vc = SuggestionViewController(viewModel: vm, type: .MostBuyed, nibName: "SuggestionViewController", bundle: nil)
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: như trên

File: MyApp/View/Controllers/Suggestion/SuggestionViewController.swift [hunk 4]
```diff
@@ -7,30 +7,92 @@
...
+    var drinkButtonleadingConstraint: CGFloat = 0
+
+    enum ScreenType {
+        case Suggestion
+        case MostBuyed
+    }
+
+    var type: ScreenType = .Suggestion
+
+    init(viewModel: SuggestionViewModel, type: ScreenType, nibName: String?, bundle: Bundle?) {
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Suggestion/SuggestionViewController.swift]: ko cần

**Pull request merged**